plugins/virtual_event/services/VirtualEventService.php

```diff
@@ -28,15 +28,27 @@ public function initService($serviceId, $serviceName, $actionName)
 	 * @param KalturaVirtualEvent $virtualEvent
 	 * @return KalturaVirtualEvent
 	 *
+	 * @throws KalturaAPIException
+	 * @throws PropelException
 	 */
 	public function addAction(KalturaVirtualEvent $virtualEvent )
 	{
-		/* @var $dbVirtualEvent VirtualEvent */
-		$this->validateScheduleEvents($virtualEvent);
-		$this->validateGroups($virtualEvent);
-		$dbVirtualEvent = $virtualEvent->toInsertableObject();
-		$dbVirtualEvent->setPartnerId($this->getPartnerId());
-		$dbVirtualEvent->save();
+		try
```

Comment:
Once you do the fix I mentioned in setRegistrationFormSchema you won't need this try catch here as well as you will directly throw an API exceptions if the string is not a valid json